indico/modules/designer/forms.py

```diff
@@ -16,15 +16,25 @@
 
 from __future__ import unicode_literals
 
-from wtforms import StringField
+from wtforms import StringField, BooleanField
 from wtforms.validators import DataRequired
 
 from indico.modules.designer.models.templates import TemplateType
+from indico.modules.events.models.events import Event
 from indico.util.i18n import _
 from indico.web.forms.base import IndicoForm
 from indico.web.forms.fields import IndicoEnumSelectField
+from indico.web.forms.widgets import SwitchWidget
 
 
 class AddTemplateForm(IndicoForm):
     title = StringField(_('Title'), [DataRequired()])
     type = IndicoEnumSelectField(_('Template'), enum=TemplateType, default=TemplateType.poster)
+    is_clonable = BooleanField(_('Allow cloning'), widget=SwitchWidget(), default=True,
+                               description=_("Allow event managers to clone this template."))
+
+    def __init__(self,  *args, **kwargs):
+        target = kwargs.pop('target')
+        super(AddTemplateForm, self).__init__(*args, **kwargs)
+        if isinstance(target, Event):
+            del self.is_clonable
```

Comment:
Like this the logic is in one place (RHs) - you need to check it once to determine whether to remove the field or not and then again when saving to determine whether to set the value on the model or not.